balsamiq.md

```diff
@@ -0,0 +1,222 @@
+    <div class="list-group">
+      {{ range first $rowCount (after (add (mul $rowCount 2) 3) .Data.Pages.ByWeight) }}
+      <a href="{{ .Permalink }}" class="list-group-item">{{ .Title }}</a>
+      {{ end }}
+    </div>
+  </div>
+</div>
+```
+
+The variable called `$featuredRows` helps determine where the count should start for columns. From there Hugo counts the number of remaining articles and divides that number by 3 (rounded up to the nearest whole number). It then creates the number of items in each column so that they are as even as possible. It takes a bit of math to do it, but that's what computers are good at anyway.
+
+And now to that "Getting Started" section at the top...
+
+![](https://media.balsamiq.com/images/docslikecode/desktop-toc-getting-started.png)
+
+I wanted different images for each of the featured articles, but I just didn't like the idea of hard-coding any links or resources, in case we decided to change things later (and also because I'm stubborn). So, the top part of the code identifies the first three articles (ordered by weight), lists their names, links them up, and then grabs an image for each of them using the following code:
+
+```html
+<img src="http://media.balsamiq.com/img/support/docs/m4d/b3/toc-button-{{ .File.BaseFileName }}.svg">
+```
+
+The trick is that the last part of the image file name is the same as the name of the markdown file. So, the article called `intro.md` uses an image called `toc-button-intro.svg`. This means that if we want other articles to show up in the "Getting Started" section we don't need to touch this page. We just adjust the weights in the front matter and add a new image that corresponds to the file that will get moved into that area.
+
+Voila!
+
+## The moral of the story
+
+So, what did we learn from these challenges? I think the most important thing is that all of them were overcome without making life harder for the content writers. We didn't compromise on keeping the workflow simple when we added functionality, even though we were tempted to. 
+
+We *could* have switched away from Markdown. We *could* have started writing HTML inside our Markdown files. We *could* have required manual template updates when we reordered articles. It took a lot of work not to do these things. But that's the beauty of programming: it's an up front investment that saves time and effort in the long run.
+
+Static sites may seem to have more limitations than the old way of doing things. But if you can find a way around the limitations, you can reap the benefits that made static sites attractive in the first place. Markdown is easy. Using GitHub is easy. Running a build command from a terminal is easy. It's writing documentation that's hard. Fortunately, that's what technical writers are good at. Having a developer liaison for the docs team can free writers from having to think about the limitations of the technology they're using so they can focus on writing the docs.
```

Comment:
I really like these suggestions. And smart to add "excellent". Writing documentation is easy. Writing _good_ documentation is hard.